common/flatpak-installation.c

```diff
@@ -1113,9 +1116,15 @@ flatpak_installation_list_installed_refs_for_update (FlatpakInstallation *self,
        * This makes sure that the ref (maybe an app or runtime) remains in usable
        * state and fixes itself through an update.
        */
-      state = flatpak_dir_get_remote_state_optional (dir, remote_name, FALSE, cancellable, error);
+      state = g_hash_table_lookup (remote_states, remote_name);
       if (state == NULL)
-        continue;
+        {
+          state = flatpak_dir_get_remote_state_optional (dir, remote_name, FALSE, cancellable, error);
+          if (state == NULL)
+            continue;
+
+          g_hash_table_insert (remote_states, g_strdup (remote_name), flatpak_remote_state_ref (state));
```

Comment:
Is the refcounting for the state objects right here?

We give a new ref to the hash table, but we don't unref the one we get from flatpak_dir_get_remote_state_optional

Reply:
Correct. Fixed now.